tests/kernel/context/src/main.c

```diff
@@ -1161,6 +1161,27 @@ void test_kernel_thread(void)
 
 }
 
+void test_kernel_cpu_idle_irq_key(void)
+{
+	unsigned int key = 0, mask = 0xF00;
+
+	key = irq_lock();
+	zassert_true((key & mask) == 0x200, "The key value is not expected");
+	k_cpu_atomic_idle(key);
+	key = irq_lock();
+	zassert_true((key & mask) == 0x200, "The key value is not expected");
+
+	/* locked irq first */
+	irq_lock();
+
+	key = irq_lock();
+	zassert_true((key & mask) == 0x0, "The key value is not expected");
```

Comment:
I think this should be:  zassert_false((key & mask), "The key value is not expected");